### PR TITLE
Migrate variant reference resolver

### PR DESCRIPTION
**Pull request by salander85**
Migration of VariantReferenceResolver to java-sdk-v2.

### CODE REVIEW

File: src/main/java/com/commercetools/sync/sdk2/commons/utils/CompletableFutureUtils.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+
+public final class CompletableFutureUtils {
```
Comment (lojzatran) [inline src/main/java/com/commercetools/sync/sdk2/commons/utils/CompletableFutureUtils.java:16]: This class was just added in v14.0.0 of java sdk v2: https://github.com/commercetools/commercetools-sdk-java-v2/pull/430

Please remove this class and use the one from the sdk.

File: src/main/java/com/commercetools/sync/sdk2/commons/utils/ResourceIdentifierUtils.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import static java.util.Optional.ofNullable;
+
+import com.commercetools.api.models.ResourceIdentifiable;
+import com.commercetools.api.models.common.Reference;
+import com.commercetools.api.models.common.ResourceIdentifier;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public final class ResourceIdentifierUtils {
```
Comment (lojzatran) [inline src/main/java/com/commercetools/sync/sdk2/commons/utils/ResourceIdentifierUtils.java:11]: I saw that I missed to migrate unit tests for this class. Can you migrate it? 🙏

File: src/main/java/com/commercetools/sync/sdk2/commons/utils/StreamUtils.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.commercetools.sync.sdk2.commons.utils;
+
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+
+public final class StreamUtils {
```
Comment (lojzatran) [inline src/main/java/com/commercetools/sync/sdk2/commons/utils/StreamUtils.java:8]: I saw that I missed to migrate unit tests for this class. Can you migrate it? 🙏

**Review by lojzatran — APPROVED** [review]
LGTM 🥳  There are just some small improvements that could be done - you can do them here or create a JIRA if you don't have time.

File: src/main/java/com/commercetools/sync/sdk2/commons/utils/CompletableFutureUtils.java (hunk 1 above)
Comment (salander85) [reply src/main/java/com/commercetools/sync/sdk2/commons/utils/CompletableFutureUtils.java:16]: https://commercetools.atlassian.net/browse/DEVX-58

**Pull request merged**